paymentsheet/src/main/java/com/stripe/android/customersheet/injection/CustomerSheetViewModelModule.kt

```diff
@@ -78,9 +80,20 @@ internal class CustomerSheetViewModelModule {
         LocaleListCompat.getAdjustedDefault().takeUnless { it.isEmpty }?.get(0)
 
     @Provides
-    fun initialViewState(paymentConfiguration: PaymentConfiguration): CustomerSheetViewState {
+    @Named(IS_LIVE_MODE)
+    fun isLiveMode(paymentConfiguration: PaymentConfiguration): Boolean {
+        return paymentConfiguration.publishableKey.startsWith("pk_live")
+    }
+
+    @Provides
+    fun initialViewState(
+        @Named(IS_LIVE_MODE) isLiveMode: Boolean
+    ): CustomerSheetViewState {
         return CustomerSheetViewState.Loading(
-            isLiveMode = paymentConfiguration.publishableKey.startsWith("pk_live")
+            isLiveMode = isLiveMode
         )
     }
+
+    @Provides
+    fun initialBackstack(): Stack<CustomerSheetViewState> = Stack()
```

Comment:
I don’t think this makes a lot of sense.

Reply:
This was done for testing purposes, I can look into a different solution. Same reason why we inject the initial view state.